Makefile

```diff
@@ -191,8 +191,12 @@ CODESPELL_VENV = $(VENV)/bin/codespell
 MDFORMAT_VENV = $(VENV)/bin/mdformat
 MKDOCS_VENV = $(VENV)/bin/mkdocs
 
+# Check that the Python version meets requirements from pyproject.toml
+_python_version_check:
+	@python3 scripts/check_python_version.py
+
 # Make a virtual environment.
-$(VENV):
+$(VENV): _python_version_check
```

Comment:
Hmm there's an unintended side effect of adding this dependency. The "Creating virtual environment" rule gets executed every time now.

<img width="534" height="158" alt="Image" src="https://github.com/user-attachments/assets/aa3091ab-1e66-4060-9b9f-84e0748fee95" />

With this in mind, let's do the following:

* Delete the `_python_version_check` rule.
* Add `@python3 scripts/check_python_version.py` immediately above the `@echo "Creating virtual environment"` line.
* Add `@python3 scripts/check_python_version.py` immediately above the `@$(PYTHON_VENV) -m uv pip install` line in the `_pyspec` rule below. I'd like to check the version there as well.